### DISCUSSION TITLE
list index out of range

### DISCUSSION THREAD

**Discussion by Sep24thLegend** [magicpdf/Magic-PDF#960] [Q&A]
![image](https://github.com/user-attachments/assets/19394b73-bd01-4f53-a7d4-5a677111e93e)
求问这种情况是为什么呢，如何解决在解析过程中遇到的这个问题

**Comment by myhloli** [comment]
可以截下报错堆栈的最下方几条或者上传一下出错的pdf文件吗？

**Reply by Sep24thLegend** [reply]
报错堆栈的最下方几条没截图，最终显示的报错貌似就是list index out of range，表格识别模型当时用的tablemaster，速度很慢。换了struct_eqtable模型后在解析过程中会出现“ table recognition processing fails, not found expected HTML table end”的警告是怎么回事呢，如下图所示
![image](https://github.com/user-attachments/assets/4e54e684-bb3f-4dee-8312-020c41f18ef1)
。换了最新接入的rapid_table模型，速度确实快了很多，但它的解析效果有的地方好像不太好，pdf中的有些表格并没有识别，如图所示
![image](https://github.com/user-attachments/assets/e9189f9c-ed75-406f-8e6d-6f2c4a431c8b)，
![image](https://github.com/user-attachments/assets/9897e844-4bc7-462f-b19d-e7706ae2139d)
也会出现连续的多表格被识别为一个表格的状况，如图所示：
![image](https://github.com/user-attachments/assets/97f956fc-7f2a-4871-b1a8-eab89fc3e42d)

**Reply by Sep24thLegend** [reply]
解析效果不太理想的pdf文件有82MB，但这里只能上传不超过25MB的pdf文件
![image](https://github.com/user-attachments/assets/b424d127-2bbf-47df-a09b-619027ef1f8d)

**Reply by Sep24thLegend** [reply]
貌似过于苛刻了，多谢

**Reply by myhloli** [reply]
struct_eqtable 默认是1024token，稍微长一点的表就会超出限额，目前比较推荐使用rapidtable
